sdk/spring/spring-cloud-azure-appconfiguration-config/src/main/java/com/azure/spring/cloud/appconfiguration/config/implementation/FeatureFlagClient.java

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
Looks like we're moving the logic from `AppConfigurationFeatureManagementPropertySource` to this `FeatureFlagClient`, I'm not sure why we need to do this.

Reply:
`FeatureFlagClient` is here to solve the problem of de-duplicating feature flags loaded from multiple sources. They can come from multiple selects, multiple stores, even from Snapshots. We could have used the old property source class somewhat as is if it wasn't for Snapshots.

Though I just did notice an issue, we might have a logic issue when a store fails to load and we retry from a replica. Will look into.